matcher_errors.go

```diff
@@ -0,0 +1,42 @@
+package elemental
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// ErrUnsupportedComparator is the error type that will be returned in the event that that an unsupported comparator
+// is used in the filter.
+type ErrUnsupportedComparator struct {
+	Err error
+}
+
+// Is reports whether the provided error has the same type as ErrUnsupportedComparator. This was added as part of the new
+// error handling APIs added to Go 1.13
+func (e ErrUnsupportedComparator) Is(err error) bool {
+	return reflect.TypeOf(err) == reflect.TypeOf(e)
+}
+
+// Unwrap returns the embedded error in ErrUnsupportedComparator.
+func (e ErrUnsupportedComparator) Unwrap() error {
+	return e.Err
+}
+
+func (e ErrUnsupportedComparator) Error() string {
+	return e.Err.Error()
```

Comment:
this is where you should prefix yourself

`return fmt.Sprintf("unsuported comparator: %s", e.Err)`

Reply:
mmm, then what do you do at the error site? for example:

```go
return false, &MatcherError{
	Err: ErrUnsupportedComparator{Err: fmt.Errorf("unsupported comparator %q", translateComparator(filter.Comparators()[i]))},
}
```

would seem awfully strange to just have an embedded error with:

`"unsupported comparator %q", translateComparator(filter.Comparators()[i])`

Reply:
It may be better to define an attribute on the type called "comparator" and use that to populate the error object at the error site so it reads like a book.

However, I have updated it as per your suggestion. Please let me know if that looks good!